core/audits/clickjacking-mitigation.js

```diff
@@ -0,0 +1,159 @@
+/**
+ * @license
+ * Copyright 2024 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {Audit} from './audit.js';
+import {MainResource} from '../computed/main-resource.js';
+import * as i18n from '../lib/i18n/i18n.js';
+
+const UIStrings = {
+  /** Title of a Lighthouse audit that evaluates whether the set CSP or XFO header is mitigating Clickjacking attacks. "XFO" stands for "X-Frame-Options". "CSP" stands for "Content-Security-Policy". */
+  title: 'Ensure Clickjacking mitigation through XFO or CSP.',
+  /** Description of a Lighthouse audit that evaluates whether the set CSP or XFO header is mitigating Clickjacking attacks. This is displayed after a user expands the section to see more. No character length limits. The last sentence starting with 'Learn' becomes link text to additional documentation. "XFO" stands for "X-Frame-Options". "CSP" stands for "Content-Security-Policy". */
+  description: 'Deployment of either the X-Frame-Options or Content-Security-Policy (with the frame-ancestors directive) header will prevent Clickjacking attacks. While the XFO header is simpler to deploy, the CSP header is more flexible. [Learn more about mitigating Clickjacking with XFO and CSP](https://developer.chrome.com/docs/lighthouse/best-practices/clickjacking-mitigation).',
```

Comment:
```suggestion
  description: 'The `X-Frame-Options` (XFO) header or the `frame-ancestors` directive in the `Content-Security-Policy` (CSP) header can be used to mitigate clickjacking attacks. While the XFO header is simpler to deploy, the `frame-ancestors` CSP directive is more flexible. [Learn more about mitigating clickjacking](https://developer.chrome.com/docs/lighthouse/best-practices/clickjacking-mitigation).',
```

Also what makes the XFO simpler to deploy? Seems like they have the same level of complexity to deploy (just modify header values).

Reply:
XFO offers just SAMEORIGIN and DENY, without any granularity and in the past it was easier to deploy (less compatibility concerns), given more browser support than CSP (nowadays, CSP is supported by all major browsers, so maybe "deployment" isn't the correct wording). That being said, for pure clickjacking mitigation, it makes sense to use XFO and only use CSP if more flexibility is needed (CSP should be primarily used for XSS mitigation and mixing it up might confuse some folks).